students/A_Zach/lesson03/mailroom1.py

```diff
@@ -0,0 +1,161 @@
+import sys
+
+#Functions
+#Add a donation to a person on the list
+def new_donation(inp,n,d):
+    response_donation = input("Please enter an amount to donate >")
+    quit_prog(response_donation)
+    
+    i = 0
+    
+    #iterate through the database until the requested name is found
+    while i < len(n):
+        #Add the donation to the name provided
+        if n[i] == inp.title():
+            d[i].append(float(response_donation))        
+            break
+        #Increase iteration if name has not been reached
+        i += 1
+    return
+
+#Show list of names
+def show_list(inp,n):
+    #print names of donors
+    for i in n:
+        print(i)
+    return
+    
+#Add a new name to the list
+def search_name(inp,n,d):
+    for i in n:
+        #if the name is in the list add a new donation. If not add the name and a new donation.
+        if inp.title() == i:
+            new_donation(inp,n,d)
+            #leave loop if name has been found and donation added
+            return
+    n.append(inp.title())
+    d.append([])
+    new_donation(inp,n,d)
+    return
+
+#make the program quit
+def quit_prog(p):
+    if p.lower() == 'quit':
+        print("Thank you. Goodbye!")
+        sys.exit()
+
+#generate an email1
+def email(inp,n,d):
+    #find the index of n in the database
+    i = n.index(inp.title())
+    #format email. n is name. format the donation amount (d[i])
+    #into a floating point with 2 decimal places.
+    print(f"\n\nDear {inp},\n\n\tWe appreciate your generous donations totalling ${sum(d[i]):.2f}.\n\nThank you,\nAndrew\n\n")
+    
+   
+    
+def send_thank_you(n,d):
+    inp = input("Who whould you like to thank?>")
+    
+    #call quit_prog function to end program
+    quit_prog(inp)
+    
+    #show the list of names using show_list function
+    while inp.title() == 'List':
+        show_list(inp,n)
+        inp = input("Who whould you like to thank?>")
+        quit_prog(inp)
+    
+    #Call the search_name function
+    search_name(inp,n,d)
+    
+    #call the email function
+    email(inp,n,d)
+    return
+
+#sum up the total donations for each person    
+def tot_don(d):
+    tots = []
+    i = 0
+    while i < len(d):        
+        tots.append(sum(d[i]))
+        i += 1
+    return tots
+
+#average the donations for each person
+def ave_don(d):
+    aves = []
+    i = 0
+    while i < len(d):        
+        aves.append(sum(d[i])/len(d[i]))
+        i += 1
+    return aves
```

Comment:
same comment here about how to loop over a list - this is also not very efficient.

you are looping 3 times to get total donations, average and sum: this can all be accomplished in a single loop